core/station/impl/src/services/external_canister.rs

```diff
@@ -718,32 +770,176 @@ impl ExternalCanisterService {
+                        });
+
+                    // adds the new call permissions
+                    for call in calls {
+                        let call_resource =
+                            Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                                CallExternalCanisterResourceTarget {
+                                    execution_method:
+                                        ExecutionMethodResourceTarget::ExecutionMethod(
+                                            CanisterMethod {
+                                                canister_id: external_canister.canister_id,
+                                                method_name: call.execution_method,
+                                            },
+                                        ),
+                                    validation_method: call.validation_method,
+                                },
+                            ));
+
+                        self.permission_service
+                            .edit_permission(EditPermissionOperationInput {
+                                auth_scope: Some(call.allow.auth_scope),
+                                users: Some(call.allow.users),
+                                user_groups: Some(call.allow.user_groups),
+                                resource: call_resource.clone(),
+                            })?;
+                    }
+                }
+                ExternalCanisterChangeCallPermissionsInput::RemoveByExecutionMethods(methods) => {
+                    // removes the all call permissions associated with the methods specified of the external canister
+                    for method in &methods {
+                        self.permission_repository
+                            .find_external_canister_call_permissions(&external_canister.canister_id)
+                            .iter()
+                            .filter(|permission| {
+                                if let Resource::ExternalCanister(
+                                    ExternalCanisterResourceAction::Call(
+                                        CallExternalCanisterResourceTarget {
+                                            execution_method:
+                                                ExecutionMethodResourceTarget::ExecutionMethod(
+                                                    CanisterMethod {
+                                                        canister_id: _,
+                                                        method_name,
+                                                    },
+                                                ),
+                                            validation_method: _,
+                                        },
+                                    ),
+                                ) = &permission.resource
+                                {
+                                    method_name == method
```

Comment:
Could we check if the current method name is contained in methods instead of running a loop over methods?